src/Domain/PasswordResetToken.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace BigGive\Identity\Domain;
+
+use Doctrine\ORM\Mapping as ORM;
+use Symfony\Component\Uid\Uuid;
+
+/**
+ * @ORM\Entity(repositoryClass="BigGive\Identity\Repository\PasswordResetTokenRepository")
+ * @ORM\Table(name="PasswordResetToken", indexes={
+ *     @ORM\Index(name="secret", columns={"secret"}),
+ * })
+ */
+class PasswordResetToken
+{
+    /**
+     * @ORM\Id
+     * @ORM\Column(type="uuid", unique=true)
+     * @ORM\CustomIdGenerator(class="\Symfony\Bridge\Doctrine\IdGenerator\UuidGenerator")
```

Comment:
I'm slightly nervous about the ID _being_ the secret, because

1. queries for records/ debugging/ asking about a token on Slack etc. then need to use a highly sensitive value, akin to a password, to have a way of identifying what we're talking about
2. a token that looks identical to a Person ID in format is more likely to be pasted somewhere public by accident

I think it might also be prudent to store just a hash of the token, and not the plain token, for the same reasons we do so with passwords.

Reply:
I've added a separate ID column. It may not be used for anything by the app but could be useful for manual DB queries.

Reply:
As discussed think it's probably not worth hashing the token - if we only store a hash we can't look it up by the raw value, so we'd need to include the ID as well or something in the link we send out, which makes things more complicated.